tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_ema.h

```diff
@@ -0,0 +1,215 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+
+#include "ckernel.h"
+#include "sfpi.h"
+
+/*
+ * @brief Loads the current input at row (I * 4) + J from the current tile in dst 0.
+ *
+ * @tparam I
+ * @tparam J
+ *
+ * 4 inputs each from 32 columns at the current offset are loaded into the LREG0-3 registers
+ * respectively from the current tile in dst 0.
+ * Values are expected to be bfloat16 format.
+ */
+template <uint32_t I, uint32_t J>
+sfpi_inline void _ema_load_curr_input_()
+{
+    constexpr uint32_t tile_offset    = 0;
+    constexpr uint32_t dst_reg_offset = tile_offset + (I * 32) + (4 * J);
```

Comment:
```suggestion
    constexpr uint32_t dst_reg_offset = (I * 32) + (4 * J);
```
`tile_offset` isn't used anywhere else. We could simplify the code a bit.

Reply:
Added a comment to clarify. For readability, it makes sense to show that we are loading from index 0 in the dst regs.
If you see the `_ema_store_current_input_`, you will see a similar pattern of code with a different tile offset.